### DISCUSSION TITLE
Keyboard shortcuts +/- to increment/decrement note duration swapped?

### DISCUSSION THREAD

**Discussion by helge17** [helge17/tuxguitar#73] [General]
According to the Key Binding Editor, the keyboard shortcut "+" should increment the note duration, "-" should decrement the duration.

But in fact "+" decrements the duration of the note (it increments the number of notes in the measure), "-" does the opposite.

Shouldn't this be the other way around? Or am I missing something?

**Comment by guiv42** [comment]
I think you're not missing anything, the code really looks inverted.
TGIncrementDurationAction multiplies a TGDuration.value variable by 2, but the name of this variable type is ambiguous: value is not a duration, but the inverse of a duration (e.g. 2=HALF and 4=QUARTER). So, **multiplying** *TGDuration.value* by 2 **divides** *note duration* by 2.
I think it would make sense to swap the codes of TGDecrementDurationAction and TGIncrementDurationAction.
Need to check first how these actions are used in all contexts including Android version

**Comment by guiv42** [comment]
These actions are used in fretboard dialog (TGFretBoard), with "arrow down" icon associated to "increment" action and vice-versa. So, looking at buttons just above the fretboard:
- clicking on "arrow up" icon increases note duration (calling TGDecrementNoteDuration)
- clicking on "arrow down" icon decreases note duration (calling TGIncrementNoteDuration)

Subjectively, I would say the *behavior* is somehow logical. Kind of a double inversion. Same thing in TGMatrixEditor and TGPiano.

In Android behavior is inverted, and seems (subjectively) counter-intuitive:
- clicking on "arrow up" icon decreases note duration (not checked in code but I guess it calls TGIncrementNoteDuration)
- clicking on "arrow down" icon increases note duration (...)

So, if it's OK for you I suggest:
- to swap the source codes of TGIncrementNoteDuration and TGDecrementNoteDuration
- to swap also the associations with icons "arrow up" and "arrow down" in TGFretBoard, TGMatrixEditor and TGPiano

This should restore a consistent behavior of "+" and "-" shortcuts, invert Android arrows behavior, and keep fretboard, piano and matrixEditor unchanged

**Comment by helge17** [comment]
Sounds correct, I will test it as soon as the pull request is here; thanks!

**Reply by guiv42** [reply]
done :)
see #93

**Comment by helge17** [comment]
Perfect, +/up is now always "increment note duration", -/down always decrements the duration. Thanks!

**Comment by guiv42** [comment]
Just a suggestion: close this discussion now it's implemented?
